### DISCUSSION TITLE
Why are flaky tests marked as failed?

### DISCUSSION THREAD

**Discussion by suuf** [cypress-io/cypress#30458] [Questions and Help]
Hello,

When there are flaky tests the Cypress/flake check in the CI/CD pipeline in Github Actions is marked as failed. This makes the entire check marked as failed. Why should this happen? 

Couldn't this be handled differently, perhaps marking the Cypress/flake check as passing, or is there a way to configure the pipeline so that it doesn’t flag flaky tests as full failures?

Screenshot:
![image](https://github.com/user-attachments/assets/e709809d-934a-4a00-8660-4f8df9e04dcf)

**Comment by portyu9** [comment]
Hi @suuf 

That behavior is intentional when Cypress Cloud's **flake alerting** status check is enabled.

A test can fail on one attempt and then pass on retry, so the final test/run can be passing while Cypress Cloud still reports that the run contained flake. The separate flake status is designed to surface that instability and can be used to block merges.

If you do not want flaky tests to block merging, you have two practical choices:

- disable flake alerting/status checks in the project's Cypress Cloud GitHub integration settings, or
- keep the flake check for visibility but do not make that check a required branch-protection status in GitHub.

That way, a pass-on-retry can still leave the normal test result green without treating the flake alert as a merge gate.

Please mark this answer as accepted if it helped, thank you.
